DSpace/DSpace
# beta4 - IP Authentication Support

## References
* Link to [JIRA](https://jira.lyrasis.org/browse/DS-4513) ticket(s)

## Description
This PR adds support for IP authentication (which grants someone rights based on groups for specific IP ranges without logging in)

## Instructions for Reviewers
In order to test this:
* Create an item/bitstream/… with permissions to a group which is not Anonymous (e.g. Staff) (there's no UI for this yet, you may need to do this in the DB directly)
* Enable IP authentication, and configure your IP with access to that group
* Verify you can access the resource
* Verify you can't access the resource from another IP

Keep in mind some complexities exist:
* This may not work as expected in the UI (Angular) if that IP doesn't have access and preboot is used to render the page. It's recommended to verify the REST queries or browse to the page in Angular
* If you change the item using the DB, discovery won't be updated and may assume the item still exists. It won't be returned thanks to https://github.com/DSpace/DSpace/pull/2726, but you may need to update discovery to ensure it behaves better
* Make sure to verify which IP is seen by DSpace, by default X-Forwarded-For headers are also supported

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/master/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
